### PR TITLE
Performance tweaks

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Removes some very common N+1s found with the bullet gem, and makes enterprises edit usable as superadmin.

Tested locally with full Aus production data.

#### What should we test?

A quick sanity check to make sure shipping methods and payment methods load correctly in enterprise edit page would be good (as superadmin and non-superadmin).

#### Release notes

Improved performance in some enterprise queries and for enterprise edit as superadmin

Changelog Category: Changed

### CODE REVIEW

**Comment by mkllnk** [comment]
@Matt-Yorkley Did you know that you can create draft pull requests?

**Comment by luisramos0** [comment]
>> @Matt-Yorkley Did you know that you can create draft pull requests?
I didnt until a few weeks ago!!! Thanks Maikel.

File: app/controllers/admin/enterprises_controller.rb [hunk 1]
```diff
@@ -32,6 +32,12 @@ def index
       end
     end
 
+    def edit
+      @object = Enterprise.where(permalink: params[:id]).
+        includes(users: [:ship_address, :bill_address]).first
+      super
+    end
+
```
Comment (luisramos0) [inline app/controllers/admin/enterprises_controller.rb]: the load_resource method in the resource controller will still execute with the N+1, right?

File: app/helpers/injection_helper.rb [hunk 2]
```diff
@@ -35,13 +35,18 @@ def inject_enterprise_shopfront_list
...
-      Enterprise.activated.visible.select(select_only).includes(address: :state).all,
+      Enterprise.activated.visible.select(select_only).includes(address: [:state, :country]).all,
       Api::EnterpriseShopfrontListSerializer
     )
   end
 
   def inject_enterprise_and_relatives
-    inject_json_ams "enterprises", current_distributor.relatives_including_self.activated.includes(address: :state).all, Api::EnterpriseSerializer, enterprise_injection_data
+    enterprises_and_relatives = current_distributor.relatives_including_self.activated.
+      includes(address: [:state, :country]).all
```
Comment (luisramos0) [inline app/helpers/injection_helper.rb]: style. I prefer to break all lines when one line is broken
```ruby
    enterprises_and_relatives = current_distributor.
         relatives_including_self.
         activated.
         includes(address: [:state, :country]).
         all
```

**Comment by sauloperez** [comment]
Moving to Test Ready but I guess you'll still fix that styling comment @luisramos0 mentioned.

File: app/controllers/admin/enterprises_controller.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/controllers/admin/enterprises_controller.rb]: No, `load_resource` starts with:
```ruby
def load_resource
  if member_action?
    @object ||= load_resource_instance

...
```
and `bullet` confirmed the N+1 was gone.

File: app/helpers/injection_helper.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply app/helpers/injection_helper.rb]: Done :+1:

File: app/controllers/admin/enterprises_controller.rb (hunk 1 above)
Comment (luisramos0) [reply app/controllers/admin/enterprises_controller.rb]: awesome!

**Comment by filipefurtad0** [comment]
Hi @Matt-Yorkley,
I checked that shipping methods and payment methods load correctly in the enterprise edit page, both as superadmin and non-superadmin.
Found nothing unusuall, so moving to ready-to-go.

**Pull request merged**